Source/Scene/ModelExperimental/GeometryPipelineStage.js

```diff
@@ -31,17 +31,16 @@ var GeometryPipelineStage = {};
  * @private
  */
 GeometryPipelineStage.process = function (renderResources, primitive) {
-  // The attribute index is taken from the node render resources, which may have added some attributes of its own.
-  var attributeIndex = renderResources.attributeIndex;
   var index;
   var setIndexedAttributeInitializationLines = [];
   var customAttributeInitializationLines = [];
   for (var i = 0; i < primitive.attributes.length; i++) {
     var attribute = primitive.attributes[i];
-    if (attribute.semantic !== VertexAttributeSemantic.POSITION) {
-      index = attributeIndex++;
-    } else {
+    if (attribute.semantic === VertexAttributeSemantic.POSITION) {
       index = 0;
+    } else {
+      // The attribute index is taken from the node render resources, which may have added some attributes of its own.
+      index = renderResources.attributeIndex++;
```

Comment:
Oh wow, nice catch, I guess we weren't updating the index in renderResources due to the local variable?

Reply:
Yeah, surprising that it did not break before.